spec/draft/design_topics/data_interchange.rst

```diff
@@ -99,4 +99,7 @@ include:
 
 There may be other reasons why it is not possible or desirable for an
 implementation to materialize the array as strided data in memory. In such
-cases, the implementation may raise a `BufferError` in the `__dlpack__` method.
+cases, the implementation may raise a `BufferError` in the `__dlpack__` or
+`__dlpack_device__` method. In case an implementation is never able to export
+its array data via DLPack, it may omit `__dlpack__` and `__dlpack_device__`
+completely, and hence `from_dlpack` may raise an `AttributeError`.
```

Comment:
Ah, a bit my bad.  In principle, I think:
* The implementor may not implement these or raise a BufferError (an `AttributeError` may occur on access).
* For `from_dlpack`: I would be OK to prescribe something actually, if @leofang thinks it is helpful.

But overall, I am happy as is.  I suspect in practice `mpi4py` will use `hasattr(obj, "__dlpack__")` or so to decide whether to try `from_dlpack` at all.

Reply:
Yes we did that in mpi4py indeed. This seems fine to me. In `from_dlpack`, perhaps we can cross-ref back to this section,  but I am fine with this as-is too.